solidity/contracts/token/libs/FeeTokenRouter.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {TokenRouter} from "./TokenRouter.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+struct Quotes {
+    address token;
+    uint256 amount;
+}
+
+/**
+ * @title Warp Route that charges non-native (ERC20) fees in addition to the native IGP fee.
+ * @author Abacus Works
+ */
+abstract contract FeeTokenRouter is TokenRouter {
+    constructor(address _mailbox) TokenRouter(_mailbox) {}
+
+    function transferRemote(
```

Comment:
I feel like we still don't have a great story for the various overrides that exist for transferRemote, i.e. for overriding the hooks and hook metadata? @yorhodes how do you think we should handle it? This is also related to a feature for warp route owners to be able to charge fees for using the warp route, but today with the approach of using `PostDispatchHooks` users can technically just override that hook

Reply:
hmm I mean I guess there are warp routes that would want to charge eg 5bps on the transfer amount but still allow overriding the hook for self relay

Reply:
although that could all be implemented in the `_transferFromSender` hook